packages/syft/src/syft/core/tensor/ancestors.py

```diff
@@ -443,9 +443,24 @@ def _private(
         #             f"Expected DataSubject to be either string or DataSubject object, but type is {type(entity)}"
         #         )
         if data_subjects.shape != self.shape:
-            raise ValueError(
-                f"DataSubjects shape: {data_subjects.shape} should match data shape: {self.shape}"
-            )
+            if data_subjects.size == 1:
+                # 1 data subject for the entire tensor.
+                data_subjects = np.broadcast_to(data_subjects, self.shape)
+            if (
+                data_subjects.shape[0] == self.shape[0]
+                and len(data_subjects.shape) == 1
+            ):
+                # e.g. 1 data subject per image for 10 imgs: child = (10, 25, 25) and data_subjects = (10,)
+                data_subjects = np.broadcast_to(data_subjects, self.shape)
```

Comment:
@IshanMi this broadcast operation will fail because data_subjects is a 1-D array. We will have to expand the data_subjects to correct the dimensions before performing the broadcast operation. Something like this should work:
```
# no of axes in the data
no_of_axes = len(self.shape)
axes_to_expand_along = list(range(no_of_axes))[1:]

# expand the data subject array to have the same dimensions as the data
data_subjects = np.expand_dims(data_subjects, axis=axes_to_expand_along)

# now we can perform the broadcast operation
data_subjects = np.broadcast_to(data_subjects, self.shape)
```